### DISCUSSION TITLE
Plan stale when applying PR that only changes the lock file

### DISCUSSION THREAD

**Discussion by florianmutter** [suzuki-shunsuke/tfaction#1529] [Deprecated (Question and Answer)] [answered]
### Question

We have a PR where we moved terraform files to a different folder. tfaction updated the lock file in the PR and plan was successful ("No changes"). After merging apply failed and said the plan is stale. It created a followup PR that we can merge and nothing fails.

Does anyone have an idea what might beging wrong here? There was no PR merged inbetween.

**Comment by suzuki-shunsuke** [comment, marked as answer]
`the lock file` means .terraform.lock.hcl, right?

> After merging apply failed and said the plan is stale.

tfaction stores the plan file at GitHub Artifacts and uses it when running `terraform apply`.

https://suzuki-shunsuke.github.io/tfaction/docs/feature/plan-file

https://github.com/suzuki-shunsuke/tfaction/blob/9d8c2351cf9530fae0f789928f2a06aeade4f697/terraform-apply/download_plan_file.sh#L1-L26

I think your failure means the plan file is stale, but basically tfaction updates plan files if necessary, so this issue won't occur.

https://suzuki-shunsuke.github.io/tfaction/docs/feature/auto-update-related-prs

If you merged pull requests regarding the same working directory at the same time,
then the issue may occur.

1. PR A is merged
2. PR B is merged
3. apply of PR A is run
4. tfaction can't update the plan file of PR B because PR B is already merged
5. apply of PR B fails because the plan file is stale

**Reply by florianmutter** [reply]
After carefully checking I noticed a run of the old pipeline between the plan creation and running apply. This was the old pipeline that we are migration away from.

Sorry for taking up your time again and thank you very much for the help
